pkg/controller/bundlec/resource_sync_task.go

```diff
@@ -497,6 +479,20 @@ func (st *resourceSyncTask) evalSpec(res *smith_v1.Resource, actual runtime.Obje
 	}
 	obj.SetOwnerReferences(refs)
 
+	// Object GroupKind-specific munging
+	switch obj.GroupVersionKind().GroupKind() {
+	case schema.GroupKind{Group: sc_v1b1.GroupName, Kind: "ServiceInstance"}:
+		obj, err = st.processServiceInstance(obj, actual, st.bundle.Namespace)
```

Comment:
Could just return there (and for deployment as well). Fine if you prefer to explicitly check error.

Reply:
Yes, I prefer that. It follows the style.